app/src/main/java/org/kiwix/kiwixmobile/language/LanguageFragment.kt

```diff
@@ -105,26 +120,27 @@ class LanguageActivity : BaseActivity() {
     Saving -> Unit
   }
 
-  override fun onCreateOptionsMenu(menu: Menu): Boolean {
-    menuInflater.inflate(R.menu.menu_language, menu)
+  override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
+    super.onCreateOptionsMenu(menu, inflater)
+    inflater.inflate(R.menu.menu_language, menu)
     val search = menu.findItem(R.id.menu_language_search)
     (search.actionView as SearchView).setOnQueryTextListener(SimpleTextListener {
       languageViewModel.actions.offer(Filter(it))
     })
-    return true
   }
 
   override fun onOptionsItemSelected(item: MenuItem): Boolean {
+    var returnValue = false
     when (item.itemId) {
```

Comment:
you can `return@onOptionsItemSelected` I think? also when can be used as an expression `var returnValue = when(...`

Reply:
ah, the issue wasn't that it returned from the when-statement but that the lint check complained about multiple return statements.